src/core/ddsi/src/ddsi_typelookup.c

```diff
@@ -280,7 +356,7 @@ void ddsi_tl_handle_request (struct ddsi_domaingv *gv, const ddsi_guid_prefix_t
       size_t sz;
       types.n++;
       types.types = ddsrt_realloc (types.types, types.n * sizeof (*types.types));
-      memcpy (&types.types[types.n - 1].type_identifier, tlm->type_id, sizeof (types.types[types.n - 1].type_identifier));
+      types.types[types.n - 1].type_identifier = tlm->type_id;
       tlm->sertype->ops->serialize (tlm->sertype, &sz, &types.types[types.n - 1].type_object.value);
```

Comment:
Error handling? I mean, it is defined as returning success/failure, and one does need to at least allow for types that do not support it.